src/api/app/controllers/webui/package_controller.rb

```diff
@@ -1129,4 +1138,19 @@ def check_build_log_access
 
     true
   end
+
+  def check_repository_and_architecture
+    @repo = @project.repositories.find_by(name: params[:repository]).try(:name)
```

Comment:
Can you call this variable `@repo_name`? because `@repo` implies that the variable stores a `Repository` object.